### PR TITLE
fix group_by compatibility with only_full_group_by in dashboard

### PR DESCRIPTION
**Pull request by masetto**

### CODE REVIEW

**Comment by masetto** [comment]
Hi @bjendres, there are some queries with `group by` clause without any aggregate functions as this (`CRM/Sepa/Page/MandateTab.php`):
```
      SELECT
        civicrm_sdd_mandate.id                                  AS mandate_id,
        civicrm_contribution_recur.id                           AS rcur_id,
        civicrm_contribution_recur.start_date                   AS start_date,
        civicrm_contribution_recur.end_date                     AS end_date,
        civicrm_contribution_recur.next_sched_contribution_date AS next_collection_date,
        last.receive_date                                       AS last_collection_date,
        last.cancel_reason                                      AS last_cancel_reason,
        civicrm_sdd_mandate.status                              AS status,
        civicrm_sdd_mandate.reference                           AS reference,
        cancel_reason.note                                      AS cancel_reason,
        civicrm_financial_type.name                             AS financial_type,
        civicrm_campaign.title                                  AS campaign,
        civicrm_sdd_mandate.reference                           AS reference,
        civicrm_contribution_recur.frequency_interval           AS frequency_interval,
        civicrm_contribution_recur.frequency_unit               AS frequency_unit,
        civicrm_contribution_recur.currency                     AS currency,
        civicrm_contribution_recur.amount                       AS amount,
        IF(civicrm_sdd_mandate.status IN ('FRST', 'RCUR', 'INIT', 'OOFF'), 'sepa-active', 'sepa-inactive')           
                                                                AS class
      FROM civicrm_sdd_mandate
      LEFT JOIN civicrm_contribution_recur ON civicrm_contribution_recur.id = civicrm_sdd_mandate.entity_id
      LEFT JOIN civicrm_financial_type     ON civicrm_financial_type.id = civicrm_contribution_recur.financial_type_id
      LEFT JOIN civicrm_campaign           ON civicrm_campaign.id = civicrm_contribution_recur.campaign_id
      LEFT JOIN civicrm_contribution last  ON last.receive_date = (SELECT MAX(receive_date) FROM civicrm_contribution
                                                                   WHERE contribution_recur_id = civicrm_contribution_recur.id
                                                                     AND contribution_status_id != 2)
      LEFT JOIN civicrm_note cancel_reason ON cancel_reason.entity_id = civicrm_contribution_recur.id
                                            AND cancel_reason.entity_table = 'civicrm_contribution_recur'
                                            AND cancel_reason.subject = 'cancel_reason'
      WHERE civicrm_sdd_mandate.contact_id = %1
        AND civicrm_sdd_mandate.type = 'RCUR'
        AND civicrm_sdd_mandate.entity_table = 'civicrm_contribution_recur'
      GROUP BY civicrm_sdd_mandate.id
      ORDER BY civicrm_contribution_recur.start_date DESC, civicrm_sdd_mandate.id DESC;"
```
probably this query can potentially return more records for the same `civicrm_sdd_mandate.id` because there are join table with many records. I'm afraid it's a bit risky for me to make changes without knowing the logic in depth....
Maybe I can make these interventions as they happen to me query errors in my db.

File: CRM/Sepa/BAO/SEPATransactionGroup.php [hunk 1]
```diff
@@ -103,8 +103,7 @@ function generateXML ($id = null) {
       LEFT JOIN civicrm_address a ON c.contact_id = a.contact_id AND a.is_primary = 1
       WHERE g.txgroup_id = %1
         AND c.contribution_status_id != 3
-        AND mandate.is_enabled = true
-      GROUP BY c.id"; //and not cancelled
+        AND mandate.is_enabled = true"; //and not cancelled
```
Comment (bjendres) [inline CRM/Sepa/BAO/SEPATransactionGroup.php:106]: just a remark: ``is_enabled`` has nothing to do with the mandate not being cancelled or not. This is (sometimes) used to keep mandates disabled until they are approved.

**Comment by bjendres** [comment]
Thanks @masetto. I'll merge and verify this after the ``1.5`` release, which is currently in beta.

**Comment by eptbertram** [comment]
@masetto you fixed it in https://github.com/Project60/org.project60.sepa/pull/633, didn't you? Wonderful, thank you!

**Comment by masetto** [comment]
@eptbertram #633 is the new pull request that completes, I think, only_full_group_by compatibility.

**Comment by bjendres** [comment]
replaced by #633 (if I'm not mistaken)

**Pull request closed without merging**